### PR TITLE
Add RouterOS 7.21 advertise-dns=self support for ipv6 nd

### PR DESCRIPTION
**Pull request by vfreex**
##### SUMMARY

RouterOS 7.21 added "self" as a valid value for the IPv6 ND advertise-dns setting. Model advertise-dns as a string enum on RouterOS 7.21 and later while preserving the previous boolean default for older versions.

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

Add tests for the version boundary and idempotence with yes, no, and self values.

```paste below

```

### CODE REVIEW

File: plugins/module_utils/_api_data.py [hunk 1]
```diff
@@ -9385,10 +9385,11 @@ def join_path(path):
                 ([('7.15', '>=')], 'comment', KeyInfo()),
                 # ([('7.15', '>=')], 'copy-from', KeyInfo(write_only=True)),
                 ([('7.15', '>=')], 'numbers', KeyInfo(read_only=True)),
+                ([('7.21', '<')], 'advertise-dns', KeyInfo(default=True)),
+                ([('7.21', '>=')], 'advertise-dns', KeyInfo(default='yes')),
```
Comment (felixfontein) [inline plugins/module_utils/_api_data.py:9389]: I don't think this change is necessary, since librouteros automatically converts the string `"yes"` to the boolean value `True`. So the current version should already work fine.

Do you have any indication that it does not work fine?

**Comment by vfreex** [comment]
Thanks for the review!  So this PR is not needed as a feature change. I’ll close it.
